### PR TITLE
fix(cdc): fix the filter expression of CdcFilter executor

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Change the filter expression to filter upstream source chunk by the value of `_rw_table_name`, which is prefixed by the schema name of the table. e.g. `public.t1`.
The previous expression only filter by table name w/o schema name.

close https://github.com/risingwavelabs/risingwave/issues/17890

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

After this PR, use needs to prefix the upstream table name with database name. e.g.
```
CREATE TABLE shared_orders (
    order_id INT,
    order_date BIGINT,
    customer_name VARCHAR,
    price DECIMAL,
    product_id INT,
    order_status SMALLINT,
    PRIMARY KEY (order_id)
)  from mssql_source table 'mydb.dbo.wrong_orders';
```

</details>

### CODE REVIEW

**Review by hzxa21 — APPROVED** [review]
LGTM. Can we also add a e2e test?

**Review by hzxa21 — APPROVED** [review]
LGTM. Thanks for the fix!

**Comment by KeXiangWang** [comment]
🤔 Will there be compatibility issue?

File: java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzChangeEventConsumer.java [hunk 1]
```diff
@@ -235,6 +235,7 @@ var record = event.value();
                         // - PG: topicPrefix.schemaName.tableName
```
Comment (KeXiangWang) [inline java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzChangeEventConsumer.java:235]: Doesn't PG have the same issue?

**Comment by StrikeW** [comment]
> 🤔 Will there be compatibility issue?

For already created sqlserver cdc table, it should be fine since it will use old implementation persisted in meta store. But new table needs to prefix the upstream table with database name.

**Comment by KeXiangWang** [comment]
Please help update the doc correspondingly. Thanks!

**Review by KeXiangWang — APPROVED** [review]

File: java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzChangeEventConsumer.java (hunk 1 above)
Comment (StrikeW) [reply java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzChangeEventConsumer.java:235]: You can see in the comment, the pg connector doesn't prefixed with database name.
`PG: topicPrefix.schemaName.tableName`

**Pull request merged**